src/django_idom/types.py

```diff
@@ -16,26 +17,40 @@
 from django.db.models.base import Model
 from django.db.models.query import QuerySet
 from django.views.generic import View
+from idom.types import Location
 from typing_extensions import ParamSpec
 
 from django_idom.defaults import _DEFAULT_QUERY_POSTPROCESSOR
 
 
-__all__ = ["_Result", "_Params", "_Data", "IdomWebsocket", "Query", "Mutation"]
+__all__ = ["_Result", "_Params", "_Data", "ComponentWebsocket", "Query", "Mutation"]
 
 _Result = TypeVar("_Result", bound=Union[Model, QuerySet[Any]])
 _Params = ParamSpec("_Params")
 _Data = TypeVar("_Data")
 
 
 @dataclass
-class IdomWebsocket:
-    """Websocket returned by the `use_websocket` hook."""
+class Connection:
```

Comment:
You can subclass core's `Connection`:

```python
from idom.types import Connection as _Connection

Connection = _Connection[ComponentWebsocket]
```

Reply:
Subclassing core's connection won't allow me to type hint `carrier` though.

Reply:
Scratch that, I see how you set it up.